streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsRebalanceListener.java

```diff
@@ -142,6 +153,19 @@ public void onPartitionsLost(final Collection<TopicPartition> lostPartitions) {
 
         Set<TaskId> lostTasks = new HashSet<>();
         final long start = time.milliseconds();
+        if (streamThread.eosBetaEnabled && !streamThread.eosUpgradeModeEnabled) {
```

Comment:
Hmm.. I think we should just close all tasks in onPartitionsLost (this is done as in https://github.com/apache/kafka/pull/7997), then we do not need to have the rebalance listener executing the embedded clients logic here.